apps/angular/ngfor-enhancement/src/app/ng-for-empty.directive.ts

```diff
@@ -0,0 +1,57 @@
+import { NgFor } from '@angular/common';
+import {
+  Directive,
+  DoCheck,
+  EmbeddedViewRef,
+  inject,
+  input,
+  TemplateRef,
+  ViewContainerRef,
+} from '@angular/core';
+
+@Directive({
+  // eslint-disable-next-line @angular-eslint/directive-selector
+  selector: '[ngForEmpty]',
+  standalone: true,
+  hostDirectives: [{ directive: NgFor, inputs: ['ngForOf:ngForEmptyOf'] }],
+})
+export class NgForEmptyDirective<T> implements DoCheck {
+  private readonly vcr = inject(ViewContainerRef);
+  readonly ngForEmptyOf = input.required<T[] | undefined>();
+  readonly ngForEmptyEmpty = input.required<TemplateRef<any>>();
+
+  private ref?: EmbeddedViewRef<unknown>;
+
+  ngDoCheck(): void {
+    this.ref?.destroy();
+
+    if (!this.ngForEmptyOf()?.length) {
+      this.ref = this.vcr.createEmbeddedView(this.ngForEmptyEmpty());
+    } else {
+      this.ref?.destroy();
+    }
+  }
+}
```

Comment:
you should use an effect here. ngDoCheck is checked at each change detection, effect will be triggered only if ngForEmptyOf is changed ?

Reply:
Interesting. I'll give that a shot.